components/password_reset_form/index.ts

```diff
@@ -1,22 +1,30 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {bindActionCreators} from 'redux';
+import {bindActionCreators, Dispatch} from 'redux';
 import {connect} from 'react-redux';
 import {getConfig} from 'mattermost-redux/selectors/entities/general';
 import {resetUserPassword} from 'mattermost-redux/actions/users';
 
+import {ServerError} from 'mattermost-redux/types/errors';
+import {GenericAction} from 'mattermost-redux/types/actions';
+import {GlobalState} from 'types/store';
 import PasswordResetForm from './password_reset_form';
 
-const mapStateToProps = (state) => {
+type Actions = {
+    resetUserPassword: (token: any, newPassword: string) => any;
```

Comment:
seems like function signature does not match the one in https://github.com/mattermost/mattermost-redux/blob/master/src/actions/users.ts#L1193
can you take a look at it?

Reply:
`token: any` was an attempt to fix this `make check-types` error.
![image](https://user-images.githubusercontent.com/10658132/98037125-f6252f80-1dcf-11eb-8b1e-8a3cf4af4c0a.png)

Let me know if there's anything else I should do to address the above error

Reply:
sorry, I missed this comment!
so the problem seems to be that on line 53 it tries to get the token from the url search params:
`        const token = (new URLSearchParams(this.props.location.search)).get('token');`
but that might not exist. if this token that is required doesn't exist, what would be the next step?

For me, it basically means that either we are doing the types wrong (we are asking it to be required but it is not) or we ar not being consistent with it (like exiting or reporting an error if the conditions doesn't match the requirements)